common/interpolation/include/interpolation/spline_interpolation.hpp

```diff
@@ -40,9 +45,86 @@ struct MultiSplineCoef
   std::vector<double> d;
 };
 
+// static spline interpolation functions
 std::vector<double> slerp(
   const std::vector<double> & base_keys, const std::vector<double> & base_values,
   const std::vector<double> & query_keys);
+
+// std::vector<double> slerpDiff(
+//   const std::vector<double> & base_keys, const std::vector<double> & base_values,
+//   const std::vector<double> & query_keys);
+
+// TODO(murooka) use template
+// template <typename T>
+// std::vector<double> slerpYawFromPoints(const std::vector<T> & points);
+std::vector<double> slerpYawFromPoints(const std::vector<geometry_msgs::msg::Point> & points);
```

Comment:
(@rej55 Let's discuss in a thread)
![image](https://user-images.githubusercontent.com/20228327/152909533-72770de8-52cd-4415-96e0-594f352cdf50.png)

Thank you. It's a good point.

For static spline interpolation (this function `slerpYawFromPoints`), I'm ok to move elsewhere like tier4_autoware_utils as you pointed.

For non-static spline interpolation, I implemented SplineInterpolation1d and SplineInterpolationPoint2d classes.
The second class includes yaw calculation (`getSplineInterpolatedYaw`). This member function is an option in this class. I mean, SplineInterpolationPoint2d is mainly for interpolating and calculating x and y, and as an option, you can get yaw angle as well (it is calculated from derivatives of x and y).
So it's not weird that SplineInterpolationPoint2d, which has `getSplineInterpolatedYaw`, is in interpolation package, IMO.

Reply:
I feel it is a little weird having `SplineInterpolation"Point2d"` here as well since it depends on the geometric configuration.

It is better to create `SplineInterpolation"Path2d"` instead and manage it in a separate file. (The interface could be the same as the "Point2d".)